server/src/com/thoughtworks/go/server/web/BackupFilter.java

```diff
@@ -0,0 +1,131 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.web;
+
+import com.google.gson.JsonObject;
+import com.thoughtworks.go.server.service.BackupService;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @understands redirecting all requests to a service unavailable page when the server is being backed up.
+ */
+public class BackupFilter implements Filter {
+
+    private FilterConfig filterConfig;
+
+    @Autowired
+    private BackupService backupService;
+
+    // For the Test
+    protected BackupFilter(BackupService backupService) {
+        this.backupService = backupService;
+    }
+
+    private String urlEncode(String string) throws UnsupportedEncodingException {
+        return URLEncoder.encode(string, "UTF-8");
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        this.filterConfig = filterConfig;
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);
+
+        String url = ((HttpServletRequest) request).getRequestURI();
+        if (isBackupFinishJsonUrl(url)) {
+            response.setContentType("application/json");
+            JsonObject json = new JsonObject();
+            json.addProperty("is_backup_finished", backupService.isBackingUp());
+            response.getWriter().print(json);
+            return;
+        }
+        if (backupService.isBackingUp()) {
+            ((HttpServletResponse) response).setHeader("Cache-Control", "private, max-age=0, no-cache");
+            ((HttpServletResponse) response).setDateHeader("Expires", 0);
+
+            if (isAPIUrl(url) && !isMessagesJson(url)) {
+                generateAPIResponse(request, response);
+            } else {
+                try {
+                    generateHTMLResponse(response);
+
+                } catch (Exception e) {
+                    e.printStackTrace();
```

Comment:
Handle this in the generate html response method and do not print stack trace, log this as error using a logger.